zipline/utils/input_validation.py

```diff
@@ -87,17 +102,12 @@ def _expect_type(type_):
     else:
         template = _template.format(type_or_types=_qualified_name(type_))
 
-    def _check_type(func, argname, argvalue):
-        if not isinstance(argvalue, type_):
-            raise TypeError(
-                template.format(
-                    funcname=_qualified_name(func),
-                    argname=argname,
-                    actual=_qualified_name(type(argvalue)),
-                )
-            )
-        return argvalue
-    return _check_type
+    return _mk_check(
+        TypeError,
+        template,
+        complement(flip(isinstance, type_)),
```

Comment:
This is a case where I actually think `lambda v: not isinstance(v, type_)` is much clearer.